docs/HowToDebugACrash.md

```diff
@@ -161,3 +162,110 @@ Callstack from values in stackdump
 0e002495	os/kernel/binary_manager/binary_manager_load.c:392
 0e001b53	os/kernel/task/task_start.c:180
 ```
+
+## Understanding Debug messages during a fault
+A crash can occur due to any one of the faults, such as, memory fault, usage fault, hard fault, etc.. When a fault occurs, the hardware encodes the cause of the fault in a corresponding fault status register. The fault handler prints the contents of this register in hexadecimal format and also provides a brief description for the cause of the fault. The following sections provide additional information and tips to understand and debug each of the fault scenarios. Here we discuss the various fault scenarios encoded in the MMFSR, BFSR, UFSR and HFSR registers for memfault, busfault, usage fault and hard fault respectively. Please refer to the ARM Architecture reference manual of the corresponding architecture to find the description of these registers and their bit fields.
+
+Armv7-m : https://developer.arm.com/documentation/ddi0403/eb/
+
+Armv8-m : https://developer.arm.com/documentation/ddi0553/latest
+
+### MemManage Fault Status Register (MMFSR)
+
+1. ```MMARVALID```
+
+Debug msg: Access violation occurred at address (MMFAR) : 0xAddress
+
+This message indicates that the address which caused the memfault is available to us. Check the printed address value. If the address is zero, it is probably caused due to a null pointer access. If the address value is non-zero, then the fault can the caused due to one of the following reasons. 
```

Comment:
Please change "then the fault can 'the' caused due to" --> "then the fault can 'be' caused due to"